### PR TITLE
Flag simulator based on Application.isEditor

### PR DESCRIPTION
**Pull request by semuserable**

### CODE REVIEW

**Comment by bitsandfoxes** [comment]
Since IApplication now has a concept of IsSimulator:
https://github.com/getsentry/sentry-unity/blob/9768a0d5b47c45466c8100af0a9bae9aca29c909/src/Sentry.Unity/SentryUnityOptions.cs#L93
https://github.com/getsentry/sentry-unity/blob/9768a0d5b47c45466c8100af0a9bae9aca29c909/src/Sentry.Unity/SentryUnityOptions.cs#L111
I think those would be fixed too?

**Comment by semuserable** [comment]
Tbh, not sure about `SentryUnityOptions`, because we have `IApplication` dependency only on integrations and not on `SentryUnityOptions` construction itself.

File: src/Sentry.Unity/Integrations/IApplication.cs [hunk 1]
```diff
@@ -27,6 +28,8 @@ private ApplicationAdapter()
 
         public string ActiveSceneName => SceneManager.GetActiveScene().name;
 
+        public bool IsSimulator => Application.isEditor;
```
Comment (bruno-garcia) [inline src/Sentry.Unity/Integrations/IApplication.cs:31]: Lets map the IApplication 1 to 1 with the Application class from Unity

```suggestion
        public bool IsEditor => Application.isEditor;

```

File: src/Sentry.Unity/UnityEventProcessor.cs [hunk 2]
```diff
@@ -20,6 +21,13 @@ internal static class UnitySdkInfo
 
     internal class UnityEventProcessor : ISentryEventProcessor
     {
+        private readonly IApplication _appDomain;
+
+        public UnityEventProcessor(IApplication? appDomain = null)
```
Comment (bruno-garcia) [inline src/Sentry.Unity/UnityEventProcessor.cs:26]: there's no AppDomain here:

```suggestion
        public UnityEventProcessor(IApplication? application = null)

```

**Comment by bruno-garcia** [comment]
> Tbh, not sure about `SentryUnityOptions`, because we have `IApplication` dependency only on integrations and not on `SentryUnityOptions` construction itself.

Ideally we'd decouple those but on a follow up PR woudd be fine. That's not high priority given we're aiming at shipping 0.1.0 asap

**Pull request merged**